crypto/evp/evp_lib.c

```diff
@@ -298,6 +298,14 @@ int EVP_CIPHER_CTX_nid(const EVP_CIPHER_CTX *ctx)
 
 int EVP_MD_block_size(const EVP_MD *md)
```

Comment:
So I gotta wonder, who needs this function? For what purpose?

Reply:
Well it is certainly used by PKIX-SSH, because that's how I knew we needed this fix (see #8565). As to why its used - perhaps @petrovr can answer.

Reply:
Ah, HMAC needs to know, and we can't assume that HMAC is provided by the same provider.
OK then!